src/js/azdata/azdataQueryPlan.js

```diff
@@ -138,18 +138,78 @@ azdataQueryPlan.prototype.init = function (container, iconPaths) {
 };
 
 azdataQueryPlan.prototype.registerZoomInListener = function (element, eventType) {
-    this.graph.addZoomInListener(element, eventType);
+    const zoomIn = () => {
+        this.graph.zoomIn();
+    };
+    this.graph.addDomEventListener(element, eventType, zoomIn);
 };
 
 azdataQueryPlan.prototype.registerZoomOutListener = function (element, eventType) {
-    this.graph.addZoomOutListener(element, eventType);
+    const zoomOut = () => {
+        this.graph.zoomOut();
+    };
+
+    this.graph.addDomEventListener(element, eventType, zoomOut);
+};
+
+azdataQueryPlan.prototype.registerZoomToFitListener = function (element, eventType) {
+    const zoomToFit = () => {
+        this.graph.fit();
+        this.graph.view.rendering = true;
+        this.graph.refresh();
+    };
+    
+    this.graph.addDomEventListener(element, eventType, zoomToFit);
 };
 
 azdataQueryPlan.prototype.registerGraphCallback = function (eventType, callback) {
     this.graph.addListener(eventType, (sender, event) => {
         this.graph.graphEventHandler(sender, event, callback);
     });
-}
+};
+
+azdataQueryPlan.prototype.getZoomLevelPercentage = function() {
+    return this.graph.view.scale * 100;
+};
+
+azdataQueryPlan.prototype.zoomTo = function (zoomPercentage) {
+    const ZOOM_PERCENTAGE_MINIMUM = 3;
```

Comment:
Why  are we limiting custom zoom to these values?  BTW, the same limitations do not apply to zoomIn and zoomOut functions? Either we should keep the limits everywhere or none at all

Reply:
I'll get rid of the upper limit. The lower limit came from seeing the zoom out button stopping when a zoom level of 3 is reached. 
![image](https://user-images.githubusercontent.com/87730006/152030626-ece5130b-1a78-4883-b8be-b532b52c4730.png)

The lower limit also helps to prevent zoom logic from breaking the graph and zoom logic when 0 is entered because the vertices and edges disappear, but not the labels.
![image](https://user-images.githubusercontent.com/87730006/152032505-3203737f-e2e1-4e66-968e-8855dd035e57.png)

Entering another zoom level after that results in NaN
![image](https://user-images.githubusercontent.com/87730006/152033364-0e685ee3-6ed1-4d09-baca-c527e62b1ccd.png)